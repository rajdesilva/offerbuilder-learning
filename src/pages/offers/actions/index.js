export { newOfferActions } from './newOfferActions';
export { marketingActions } from './marketingActions';
export { offerListActions } from './offerListActions';
export { offerListSearchAndFilterActions } from './offerListSearchAndFilterActions';