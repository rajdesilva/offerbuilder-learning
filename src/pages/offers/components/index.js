export { default as CreateNewOffer } from "./CreateNewOffer";
export { default as OfferList } from "./OfferList";
export { default as Settings } from "./Settings";
export { default as Marketing } from "./Marketing";
export { default as Descriptions } from "./Descriptions";
export { default as Images } from "./Images";
export { default as LanguageSettingsModal } from "./LanguageSettingsModal";
export { default as DescriptionTabs } from "./DescriptionTabs";
export { default as PropertyMarketingInfoList } from "./PropertyMarketingInfoList";
export { default as OfferCreatedSuccess } from "./OfferCreatedSuccess";
export { default as ActiveOffers } from "./ActiveOffers";
export { default as ActiveOffersPagination } from "./ActiveOffersPagination";
export { default as ArchiveOffers } from "./ArchiveOffers";
export { default as ArchiveOffersPagination } from "./ArchiveOffersPagination";
export { default as ChangeOfferStatusModal } from "./ChangeOfferStatusModal";
export { default as OfferEdit } from "./OfferEdit";
export { default as OfferFilters } from "./OfferFilters";
export { default as PropertyImageSelectionModal } from "./PropertyImageSelectionModal";
export { default as ViewDescriptions } from "./ViewDescriptions";
export { default as ViewDeepLinkSettings } from "./ViewDeepLinkSettings";
export { default as ViewDescriptionTabs } from "./ViewDescriptionTabs";
export { default as ViewMarketing } from "./ViewMarketing";
export { default as ViewImages } from "./ViewImages";
export { default as ViewOffer } from "./ViewOffer";
export { default as ViewSettings } from "./ViewSettings";
export { default as ViewPropertyMarketingInfoList } from "./ViewPropertyMarketingInfoList";
export { default as ViewTarget } from "./ViewTarget";
export { default as OfferImages } from "./OfferImages";
export { default as OfferImageGallery } from "./OfferImageGallery";
export { default as OfferImageSelectionModal } from "./OfferImageSelectionModal";
export { default as OfferImageFilter } from "./OfferImageFilter";
export { default as UploadImage } from "./UploadImage";
export { default as OfferSelectedImages } from "./OfferSelectedImages";
export { default as OfferImageDetails } from "./OfferImageDetails";
export { default as OfferImagesPagination } from "./OfferImagesPagination";
export { default as OfferImagesCarousel } from "./OfferImagesCarousel";
export { default as PropertiesListItem } from "./PropertiesListItem";
export { default as PropertyDetailInfoPage } from "./PropertyDetailInfoPage";
export { default as PropertyImageCarousel } from "./PropertyImageCarousel";
export { default as PropertySelectionCarousel } from "./PropertySelectionCarousel";
export { default as PropertyImagesSelected } from "./PropertyImagesSelected";
export { default as PropertyDetailsImageModal } from "./PropertyDetailsImageModal";
export { default as PropertyImageListPagination } from "./PropertyImageListPagination";
