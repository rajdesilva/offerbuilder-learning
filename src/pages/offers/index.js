export {
  getLanguages,
  updateOffer,
  deleteOffer,
  getOfferDetails,
  saveEditedOffer,
  searchAndFilterOffer,
  getCloudinarySignature,
  getCityFromLatLong,
  getImagesFromProperties,
  searchAndFilterImages,
  uploadImage,
} from "./service";
