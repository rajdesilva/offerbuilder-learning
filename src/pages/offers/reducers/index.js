export { newOfferSettingParams } from './newOfferSettings';
export { languageInfo } from './languages';
export { newOfferMarketingInfo } from './newOfferMarketingInfo';
export { offerListSearchAndFilters } from './offerListSearchAndFilters';
export { offerImageSearchAndFilters } from "./offerImageSearchAndFilters";