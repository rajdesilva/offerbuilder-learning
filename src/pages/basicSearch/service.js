import { browseSupplyActions, supplySearchActions } from "./actions";
import { store } from "./../../redux/store";
import { globalActions } from "../../redux/globalActions";
import { baseURL } from "../../helpers";
import {
  createSearchQueryURL,
  getSearchParamsToFetchMarketPrice,
} from "../../helpers/utility";
import { offerListSearchAndFilterActions } from "../offers/actions";

export const searchSupply = (routePostSearch, isEditFlow) => {
  store.dispatch({
    type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_LOADING,
    payload: true,
  });
  store.dispatch({
    type: supplySearchActions.SET_INCLUDE_MARKET_PRICE_FLAG,
    payload: false,
  });

  window
    .fetchWrapper(`${baseURL.API_URL_BASE}search/` + createSearchQueryURL(routePostSearch))
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_SUCCESS,
          payload: window.getValue(response, "data.properties"),
          totalMatch: window.getValue(response, "data.totalMatchedProperties"),
          pageOffset: window.getValue(response, "data.pageOffset"),
          pageSize: window.getValue(response, "data.pageSize"),
        });
      } else {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
          payload: window.getValue(response, "error.apierror.message"),
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
        payload: error.toString(),
      });
      store.dispatch({
        type: globalActions.SET_ERROR,
        payload: error.toString(),
      });
    });
};

export const advancedSearch = (
  params,
  searchType,
  page,
  pageSize,
  sortBy,
  order,
  groupType
  ) => {
    return window.fetchWrapper(`${baseURL.API_URL_BASE}global/properties?address=${params.address}&city=${params.city}&countryCode=${params.countryCode}&distance=${params.distance}&hotelName=${params.hotelName}&latitude=${params.latitude}&longitude=${params.longitude}&zip=${params.zip}&pageOffset=${page}&pageSize=${pageSize}&propertyGroupType=${groupType}&sortCriteria=${sortBy}&sortOrder=${order}&globalSearchType=${searchType}`);

    //return window.fetchWrapper(`https://offerbuilder-api-dev.kognitiv.com/global/properties?distance=10&globalSearchType=BASIC&latitude=47.2816&longitude=15.48809&propertyGroupType=ALL`);
}

export const globalSearchResult =(params)=>{
  
  window
    .fetchWrapper(`/global/properties?address=${params.address}`)
    .then((response) => {
        console.log(response)
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
}

export const getChannels = () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_CHANNELS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}portals`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_SUCCESS,
          payload: window.getValue(response, "data.result"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getBrands = async () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_BRANDS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}sources`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_SUCCESS,
          payload: window.getValue(response, "data.brands"),
        });
        store.dispatch({
          type: offerListSearchAndFilterActions.UPDATE_OFFER_SELECTORS,
          payload: window.getValue(response, "data.brands"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_FAILURE,
          payload: [],
        });
      }
      return window.getValue(response, "data.brands");
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_BRANDS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getCurrencies = async () => {
  store.dispatch({
    type: supplySearchActions.FETCH_CURRENCIES_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}currencies`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_SUCCESS,
          payload: window.getValue(response, "data.currencies"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getMarketPricesForListOfProperties = () => {
  store.dispatch({
    type: supplySearchActions.FETCH_MARKET_PRICE_LOADING,
    payload: true,
  });

  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}search/marketprice/` +
        getSearchParamsToFetchMarketPrice()
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
        payload: [],
      });
      return [];
    });
};
