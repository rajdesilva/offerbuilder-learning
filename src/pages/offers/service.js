import { baseURL } from "../../helpers";
import {
  getParsedOffers,
  getSearchAndFilterImagesQueryUrl,
  getSearchAndFilterQueryURL,
  updateReduxWithOfferData,
} from "../../helpers/utility";
import { store } from "../../redux/store";
import {
  marketingActions,
  newOfferActions,
  offerListSearchAndFilterActions,
} from "./actions";
import { offerImageSearchAndFilterActions } from "./actions/offerImageSearchAndFilterActions";

export const validateOfferId = (offerId) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}/exist`)
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error.toString();
    });
};

export const getCloudinarySignature = () => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}cloudinary/signature`)
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error.toString();
    });
};

export const getCityFromLatLong = (latLong) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latLong.lat},${latLong.lng}&sensor=true&key=AIzaSyDydZJ7RZF1SAXwg4pZqnFFskYL8CW5RqA`;
  return fetch(url)
    .then((response) => response.json())
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error;
    });
};

export const getLanguages = () => {
  store.dispatch({
    type: marketingActions.FETCH_LANGUAGE_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}languages`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: marketingActions.FETCH_LANGUAGE_SUCCESS,
          payload: window.getValue(response, "data.languages"),
        });
      } else {
        store.dispatch({
          type: marketingActions.FETCH_LANGUAGE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: marketingActions.FETCH_LANGUAGE_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const createNewOffer = (newOfferInfo) => {
  const url = `${baseURL.API_URL_BASE}offers`;
  store.dispatch({
    type: newOfferActions.CREATE_NEW_OFFER_LOADING,
    payload: true,
  });
  return window
    .fetchWrapper(url, {
      method: "POST",
      body: JSON.stringify(newOfferInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const saveEditedOffer = (updatedOfferInfo) => {
  const url = `${baseURL.API_URL_BASE}offers/${updatedOfferInfo.id}`;
  return window
    .fetchWrapper(url, {
      method: "PATCH",
      body: JSON.stringify(updatedOfferInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const getOfferDetails = (offerId) => {
  store.dispatch({
    type: offerListSearchAndFilterActions.OFFER_DETAILS_LOADING,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}`)
    .then((response) => {
      if (response.success) {
        let offerData = window.getValue(response, "data.offer");
        offerData = getParsedOffers(offerData);
        updateReduxWithOfferData(offerData);
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_DETAILS_SUCCESS,
        });
      } else {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_DETAILS_FAILURE,
        });
      }
    })
    .catch((error) => {
      console.log(error.toString());
      store.dispatch({
        type: offerListSearchAndFilterActions.OFFER_DETAILS_FAILURE,
      });
    });
};

export const deleteOffer = (offerId) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}`, {
      method: "DELETE",
    })
    .then((response) => response);
};

export const updateOffer = (offerId, status) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}/${status}`, {
      method: "PUT",
    })
    .then((response) => response);
};

export const searchAndFilterOffer = (offerType) => {
  store.dispatch({
    type: offerListSearchAndFilterActions.OFFER_LIST_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}offers/search/` +
        getSearchAndFilterQueryURL(offerType)
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_LIST_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_LIST_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      console.error("error =", error.toString());
      store.dispatch({
        type: offerListSearchAndFilterActions.OFFER_LIST_FAILURE,
        payload: [],
      });
    });
};

export const getImagesFromProperties = (propertyCode) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}search?propertyCode=${propertyCode}`)
    .then((response) => response);
};

export const searchAndFilterImages = () => {
  store.dispatch({
    type: offerImageSearchAndFilterActions.OFFER_IMAGE_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}images/search` +
        getSearchAndFilterImagesQueryUrl()
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: offerImageSearchAndFilterActions.OFFER_IMAGE_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: offerImageSearchAndFilterActions.OFFER_IMAGE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      console.error("error =", error.toString());
      store.dispatch({
        type: offerImageSearchAndFilterActions.OFFER_IMAGE_FAILURE,
        payload: [],
      });
    });
};

export const uploadImage = (file) => {
  let myHeaders = new Headers();

  myHeaders.append("Accept", "application/json");
  let formdata = new FormData();
  formdata.append("file", file);
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}images/upload`, {
      method: "POST",
      headers: myHeaders,
      body: formdata,
    })
    .then((response) => response);
};
