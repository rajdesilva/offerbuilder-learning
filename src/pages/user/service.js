import { baseURL } from "../../helpers";
import { userActions } from "./actions/userActions";
import { store } from "../../redux/store";

export const fetchUserInfo = () => {
  const url = `${baseURL.API_URL_BASE}user/`;
  store.dispatch({
    type: userActions.FETCH_USER_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(url)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_FAILURE,
          payload: false,
        });
      }
    })
    .catch(() => {
      store.dispatch({
        type: userActions.FETCH_USER_FAILURE,
        payload: false,
      });
    });
};

export const getUsersList = () => {
  store.dispatch({
    type: userActions.FETCH_USER_MANAGEMENT_LIST_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}users/`
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
        payload: [],
      });
    });
};

export const addNewUser = (newUserInfo) => {
  const url = `${baseURL.API_URL_BASE}users/user`;
  return window
    .fetchWrapper(url, {
      method: "POST",
      body: JSON.stringify(newUserInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const deleteUser = (email, userRole) => {
  const url = `${baseURL.API_URL_BASE}users/user/${email}?roleNames=${userRole}`;
  return window
    .fetchWrapper(url, {
      method: "DELETE",
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};
