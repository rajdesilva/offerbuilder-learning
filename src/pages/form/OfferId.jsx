import React, { Fragment, useState, useCallback } from "react";
import { Input, message, Space, Spin } from "antd";
import { validateOfferId } from "../offers/service";
import { useIntl } from "react-intl";

function OfferId(props) {
  const [loading, setLoading] = useState(false);
  const intl = useIntl();
  
  return (
    <Space direction="horizontal" size="small">
      <Input
       
        placeholder={intl.formatMessage({
          id: "nemo.mediumInput",
        })}
        maxLength={24}
        data-testid="settings-offer-id"
        
      />
      {loading ? <Spin /> : <Fragment />}
    </Space>
  );
}

export default OfferId;
