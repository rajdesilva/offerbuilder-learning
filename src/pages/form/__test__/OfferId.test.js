import React from "react";
import OfferId from "../OfferId";
import * as service from "../../offers/service";
import { act, render, fireEvent, cleanup } from "../../../helpers/testUtils";

afterEach(cleanup);

describe("OfferId component test cases", () => {
  test("OfferId components snapshot", () => {
    const props = {
      onChange: jest.fn(),
    };
    const { container } = render(<OfferId {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("validate offer id by changing input field for offer id", async () => {
    const onChangeMock = jest.fn();
    const validateOfferId = jest.spyOn(service, "validateOfferId");
    validateOfferId.mockResolvedValue({
      success: true,
      data: {
        available: true,
      },
    });
    const newOfferId = "NEWYEAR2021";
    const props = {
      onChange: onChangeMock,
      value: "YEAR2021",
    };
    const { getByTestId } = render(<OfferId {...props} />);

    await act(async () => {
      await fireEvent.change(getByTestId("settings-offer-id"), {
        target: { value: newOfferId },
      });
    });

    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(validateOfferId).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toHaveBeenCalledWith(newOfferId);
  });
  test("validate offer id by changing input field for offer id and offerid is not available from backend", async () => {
    const onChangeMock = jest.fn();
    const validateOfferId = jest.spyOn(service, "validateOfferId");
    validateOfferId.mockResolvedValue({
      success: true,
      data: {
        available: false,
      },
    });
    const newOfferId = "NEWYEAR2021";
    const props = {
      onChange: onChangeMock,
      value: "YEAR2021",
    };
    const { getByTestId } = render(<OfferId {...props} />);

    await act(async () => {
      await fireEvent.change(getByTestId("settings-offer-id"), {
        target: { value: newOfferId },
      });
    });

    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toHaveBeenCalledWith("error");
  });
});
