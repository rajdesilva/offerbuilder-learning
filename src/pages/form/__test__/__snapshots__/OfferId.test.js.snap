// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferId component test cases OfferId components snapshot 1`] = `
<div
  class="ant-space ant-space-horizontal ant-space-align-center"
>
  <div
    class="ant-space-item"
  >
    <input
      class="ant-input"
      data-testid="settings-offer-id"
      maxlength="24"
      placeholder="Enter name"
      type="text"
      value=""
    />
  </div>
</div>
`;
